.github/workflows/crds-verify-kind.yaml

```diff
@@ -35,21 +35,29 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true'
         run: |
           make local
+  # Create json of k8s versions to test
+  # from guide: https://stackoverflow.com/a/65094398/4590470
+  setup-test-matrix:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Set k8s versions
+        id: set-matrix
+        # everything excluding older tags below 1.23. limits needs to be high enough to cover all latest versions
+        # grep -E "v[1-9]\.(2[3-9]|[3-9][0-9])" filters for v1.23 to v9.99
+        run: |
+          echo "matrix={\"k8s\":$(wget -q -O - "https://hub.docker.com/v2/namespaces/kindest/repositories/node/tags?page_size=50" | grep -o '"name": *"[^"]*' | grep -o '[^"]*$' | grep -v -E "alpha|beta" | grep -E "v[1-9]\.(2[3-9]|[3-9][0-9])" | awk -F. '{if(!a[$1"."$2]++)print $1"."$2"."$NF}' | sort -r | sed s/v//g | jq -R -c -s 'split("\n")[:-1]')}" >> $GITHUB_OUTPUT
```

Comment:
Could you give more explanation and comments to this piece of code ` awk -F. '{if(!a[$1"."$2]++)print $1"."$2"."$NF}'`?

It's not easy to understand what it does.

Reply:
Another comment is we don't need to support that much of the KinD versions.
I suggest to deprecate the testing of v1.23 and v1.24.

Reply:
The awk removes older patches of the same release so only 1.30.1 runs, and not 1.30.0